src/internal/hooks/use-table-interaction-metrics/index.ts

```diff
@@ -13,26 +31,41 @@ to be the cause of the current loading state.
 const USER_ACTION_TIME_LIMIT = 1_000;
 
 export interface UseTableInteractionMetricsProps {
+  elementRef: React.RefObject<HTMLElement>;
   instanceIdentifier: string | undefined;
   loading: boolean | undefined;
   itemCount: number;
   getComponentIdentifier: () => string | undefined;
+  getComponentConfiguration: () => JSONObject;
   interactionMetadata: () => string;
 }
 
 export function useTableInteractionMetrics({
+  elementRef,
   itemCount,
   instanceIdentifier,
   getComponentIdentifier,
+  getComponentConfiguration,
   loading = false,
   interactionMetadata,
 }: UseTableInteractionMetricsProps) {
+  const taskInteractionId = useRandomId();
+  const tableInteractionAttributes = useTaskInteractionAttribute(elementRef, taskInteractionId);
   const lastUserAction = useRef<{ name: string; time: number } | null>(null);
   const capturedUserAction = useRef<string | null>(null);
   const loadingStartTime = useRef<number | null>(null);
 
-  const metadata = useRef({ itemCount, getComponentIdentifier, interactionMetadata });
-  metadata.current = { itemCount, getComponentIdentifier, interactionMetadata };
+  const metadata = useRef({ itemCount, getComponentIdentifier, getComponentConfiguration, interactionMetadata });
+  metadata.current = { itemCount, getComponentIdentifier, getComponentConfiguration, interactionMetadata };
+
+  useEffect(() => {
+    ComponentMetrics.componentMounted({
```

Comment:
With every `taskInteractionId` change, do we wanna track mount?

Reply:
I think as long as it's a ref it should be fine, but I guess that also relies on the assumption that `useRandomId` is always going to return a ref.